R/generateFilterValues.R

```diff
@@ -125,22 +125,29 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
     }
 
   } else {
-    fval = lapply(filter, function(x) {
-      x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[x$name]]))
+    index_names = names(method)
+    if (is.null(index_names)) {
+      index_names = method
+    }
+    fval = mapply(function(x, name) {
+      x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[name]]))
       missing.score = setdiff(fn, names(x))
       x[missing.score] = NA_real_
       x[match(fn, names(x))]
-    })
+    }, filter, index_names, SIMPLIFY = FALSE)
     fval = do.call(cbind, fval)
     colnames(fval) = method
+    if (!is.null(names(method))) {
+      colnames(fval) = names(method)
+    }
     types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
 
     out = data.table(name = row.names(fval),
       type = types, fval, row.names = NULL, stringsAsFactors = FALSE)
 
     # variable.factor = FALSE has no effect
-    out = melt(out, value.name = "value", measure.vars = method,
-      variable.name = "filter")
+    out = melt(out, value.name = "value", measure.vars = colnames(fval),
+      variable.name = "method")
```

Comment:
@annette987 This change caused the test failures. It probably slipped while solving the merge conflicts in master.

Reply:
Actually that change was deliberate and was there before the merge. The reason for it is that method can now have duplicate filter names, which will cause the melt to fail. So I used colnames(fval) because that is set to names(method) if not null and method otherwise. In fact I could have used index_names, which is set to teh same thing.

Reply:
It could be that this change is still needed then. However, in one of the latest PRs there was the change to `variable.name = "filter"` which definitely needs to come into this PR here.

(The column name listing all the methods used is no longer called "method" but "filter" now to avoid name clashes in other places.)

If tests fail you should then know why and be able to solve the issue quickly :)

Reply:
All OK now. It was just the change to variable.name = "filter" as you suggested. I think the test I already added should be sufficient: https://github.com/annette987/mlr/blob/21c3b8b768baa0b1d7fcad5faf1bb38d157d7fbc/tests/testthat/test_featsel_filters.R#L149
Thanks for your patience with a Github newbie! :-)

Reply:
Thanks for your patience - we know how hard it is to make PRs to packages to comply with everything and especially in mlr 😄 

It seems that you need to update `man/` again, just run `devtools::document()´ or press CTRL+Shift+D in case you are using RStudio.

Reply:
:relaxed:
I ran devtools::document() and it has changed a huge number of files, not just the two with conflicts. Is that OK to push them all?

Reply:
I'd assume that you are not running the latest version of _roxygen_?
When developing pkgs, it is important to always have the most recent versions of pkgs - we usually update every few days :)

Reply:
Oh OK. I will do that from now on. Thanks.